.github/workflows/r_tests.yml

```
@@ -0,0 +1,41 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+#
+# See https://github.com/r-lib/actions/tree/master/examples#readme for
+# additional example workflows available for the R community.
+
+name: R
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        r-version: ['3.6.3']
```

Have we good any code that is Windows-specific? Many some of the filename/path code/conventions? If so, perhaps worthwhile considering a Windows build also? Definitely increases the runtime, I'm not 100% sure on it

```suggestion
    runs-on: ${{ matrix.config.os }}
    strategy:
      matrix:
        r-version: ['3.6.3']
      config:
        config:
          - {os: macOS-latest,   r: 'release'}
          - {os: windows-latest, r: 'release'}
```

---

Apologies, I think Windows would be something like the following:

```runs-on: ${{ matrix.config.os }}
    strategy:
      matrix:
        config:
          - {os: macOS-latest,   r: '3.6.3'}
          - {os: windows-latest, r: '3.6.3'}
```

---

Good thinking, if anything this should be primarily tested on Windows since that's what almost all our users are on.

I don't think it'll take too much longer to complete since the jobs are run in parallel unless otherwise specified by the looks of things.

---

Perfect - great stuff!

---

Added and looks like it all ran successfully